### PR TITLE
Add tkwant package

### PR DESCRIPTION
**Pull request by thkloss**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by thkloss** [comment]
@conda-forge/staged-recipes request for feedback and merge

File: recipes/tkwant/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{% set version = "1.0.0a0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  # url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/kwant-{{ version }}.tar.gz
+  # XXX: set to correct version when ready
```
Comment (scopatz) [inline recipes/tkwant/meta.yaml:11]: Please remove comments

File: recipes/tkwant/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+{% set name = "tkwant" %}
+{% set version = "1.0.0a0" %}
```
Comment (scopatz) [inline recipes/tkwant/meta.yaml:2]: Is there a fully released, non-alpha version to package?

Comment (thkloss) [reply recipes/tkwant/meta.yaml:2]: Yes, we plan the full release of the code, documentation and also a supporting paper soon.
However, as the documentation, which includes the installation instructions, is part
of the source code, we like to have an alpha release available in order to better test
and to fine tune our documentation and the code prior to the public release.
This is why the current version is tagged as alpha, and not yet as final 1.0.0 .

File: recipes/tkwant/meta.yaml (hunk 1 above)
Comment (thkloss) [reply recipes/tkwant/meta.yaml]: The comments have been removed.

File: recipes/tkwant/meta.yaml (hunk 2 above)
Comment (thkloss) [reply recipes/tkwant/meta.yaml]: The "rc" and "dev" are breaking the PEP 440 version scheme https://www.python.org/dev/peps/pep-0440/#pre-releases
that we like to keep. Moreover, to be honest, the CFEP-05 instructions look kind of cumbersome to me.

So at the end we made the final release 1.0.0 and adapted the recipe accordingly.

Comment (thkloss) [reply recipes/tkwant/meta.yaml]: Dear @scopatz, what do you think, is it ok like this or are there some more things to adapt?

File: recipes/tkwant/build.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+platform='unknown'
+unamestr=`uname`
+if [[ "$unamestr" == 'Darwin' ]]; then
+   cp $RECIPE_DIR/build_mac.conf build.conf
+else
+   cp $RECIPE_DIR/build_linux.conf build.conf
+fi
+
+sed -i -e "s:PREFIX:$PREFIX:g" build.conf
```
Comment (scopatz) [inline recipes/tkwant/build.sh:10]: I think inplace sed replacements are slightly different on Mac. Can you double check this please?

File: recipes/tkwant/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    - pytest-runner
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - scipy
+    - pytest
+    - sympy
+    - tinyarray
+    - mpmath
+    - matplotlib
```
Comment (scopatz) [inline recipes/tkwant/meta.yaml:42]: This should be `matplotlib-base`

Comment (thkloss) [reply recipes/tkwant/meta.yaml]: Dear @scopatz , I changed matplotlib to matplotlib-base, but there is an "error: command 'gcc' failed: No such file or directory: 'gcc'" (link below). It does not seem to be related to matplotlib, and when I turn back the change and test in a container on my local machine, the same error appears. Have you seen this problem elsewhere, it is mysterious to me why the linux pipeline is not working any more, as it did 3 days before?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=203036&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=741

File: recipes/tkwant/build.sh (hunk 3 above)
Comment (thkloss) [reply recipes/tkwant/build.sh]: It seems that "sed -i'' .." is the right way.

Comment (thkloss) [reply recipes/tkwant/build.sh:10]: There is however an error on Mac with above command, if I turn it back, at least the Mac build works fine.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=203061&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=2677

File: recipes/tkwant/meta.yaml (hunk 4 above)
Comment (scopatz) [reply recipes/tkwant/meta.yaml]: Yeah that seems unrelated to matplotlib-base, CC @conda-forge/staged-recipes

Comment (thkloss) [reply recipes/tkwant/meta.yaml]: @conda-forge/staged-recipes : Running the same recipe again today the problem disappeared.

File: recipes/tkwant/build.sh (hunk 3 above)
Comment (thkloss) [reply recipes/tkwant/build.sh]: @scopatz : The changed sed placement works today as well.

File: recipes/tkwant/meta.yaml (hunk 4 above)
Comment (scopatz) [reply recipes/tkwant/meta.yaml]: Ok! This looks good now

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**